sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+                }
+    
+                KeyClause kc = new KeyClause(op, kcv);
+                kcs.add(kc);
+            }
+    
+            return new BooleanKeyExpression(kcs);
+        }
+
+        AttributeBooleanExpression constructAttributeBoolean() throws AutoConfigureException {
+            Map<String, SingleAttributeClause> prefixes = new HashMap<>();
+            List<String> sortedPrefixes = new ArrayList<>();
+            for (AttributeValueFQN aP : policy) {
+                AttributeNameFQN a = aP.prefix();
+                String p = a.toString().toLowerCase();
+                SingleAttributeClause clause = prefixes.get(p);
+                if (clause != null) {
+                    clause.values.add(aP);
+                } else if (byAttribute(aP) != null) {
+                    var x = new SingleAttributeClause(byAttribute(aP).attr, new ArrayList<AttributeValueFQN>(Arrays.asList(aP)));
+                    prefixes.put(p, x);
+                    sortedPrefixes.add(p);
+                }
+            }
+
+            List<SingleAttributeClause> must = sortedPrefixes.stream()
+                    .map(prefixes::get)
+                    .collect(Collectors.toList());
+
+            return new AttributeBooleanExpression(must);
+        }
+
+
+
+        public class AttributeMapping {
+
+            private Map<AttributeNameFQN, Attribute> dict;
+        
+            public AttributeMapping() {
+                this.dict = new HashMap<>();
+            }
+        
+            public void put(Attribute ad) throws Exception {
+                if (this.dict == null) {
+                    this.dict = new HashMap<>();
+                }
+        
+                AttributeNameFQN prefix;
+                try {
+                    prefix = new AttributeNameFQN(ad.getFqn());
+                } catch (Exception e) {
+                    throw new Exception(e);
+                }
+        
+                if (this.dict.containsKey(prefix)) {
+                    throw new Exception("Attribute prefix already found: [" + prefix.toString() + "]");
+                }
+        
+                this.dict.put(prefix, ad);
+            }
+        
+            public Attribute get(AttributeNameFQN prefix) throws Exception {
+                Attribute ad = this.dict.get(prefix);
+                if (ad == null) {
+                    throw new Exception("Unknown attribute type: [" + prefix.toString() + "], not in [" + this.dict.keySet().toString() + "]");
+                }
+                return ad;
+            }
+
+        }
+
+        public class SingleAttributeClause {
+
+            private Attribute def;
+            private List<AttributeValueFQN> values;
+        
+            public SingleAttributeClause(Attribute def, List<AttributeValueFQN> values) {
+                this.def = def;
+                this.values = values;
+            }
+        }
+
+        public class AttributeBooleanExpression {
```

Comment:
These can be package protected
```suggestion
        class AttributeBooleanExpression {
```